Geospatial data: lat and long (#248)
arcus/education_modules: geocode_lat_long/geocode_lat_long.md

<!--

author:   Elizabeth Drellich
email:    drelliche@chop.edu
version:  1.0.0
module_template_version: 3.0.0
language: en
narrator: UK English Female
title: Encoding Geospaital Data: Latitude and Longitude
comment:  This is an introduction to latitude and longitude and the importance of geocoding - encoding geopspatial data in this coordinate system.
long_description: If you use any geospatial data, such as patient or participant addresses, it is important that that location data be in a usable form. This means using the same coordinate system that Global Positioning Systems use: latitude and longitude. This module is appropriate, as either an introduction or review, for anyone considering using geospatial data in their analysis. 
estimated_time: 15 minutes

@learning_objectives  

After completion of this module, learners will be able to:

> - understand the importance of geocoding addresses

Reviewer: ```suggestion
- Understand the importance of geocoding addresses
```

> - understand the latitude and longitude coordinate system

Reviewer: ```suggestion
- Understand the latitude and longitude coordinate system
```
You know I am funny about starting things with a capital letter, but you do you, I'm willing to be overridden...

Author: I'm capital letter flexible :)

> - geocode single addresses.  

Reviewer: ```suggestion
- Geocode single addresses.  
```
Optional comment goes here, below the three tick marks

-->

# Encoding Geospaital Data: Latitude and Longitude

<div class = "overview">

## Overview
@comment

**Is this module right for me?** @long_description

**Estimated time to completion:** @estimated_time

**Pre-requisites**
None

**Learning Objectives**

@learning_objectives

</div>

## Where is it Located?

"Where is the research building?" If you have ever need to go somewhere new, you have run into the primary issue with the way humans discuss location.

Depending on who (or what) you ask, the answer could take many forms, each of which can be useful in some situations and useless in others. Consider the types of answers you could get to the question "Where is the Roberts Center for Pediatric Research?"

- **Relational Location**: "The research building is on the other side of the South St. bridge from the hospital." 

- **Directions**: "Go three blocks that way, turn left, and you will see the research building." 

- **Street Address**: "The research building is at 734 Schuylkill Ave."

- **Mailing Address**: "The research building is at 734 Schuylkill Ave. Philadelphia, PA 19146."

- **Coordinates**: "The research building is at 39.9458 North, -75.1865 West."

> 

Author: <div class = "care">
<b style="color: rgb(var(--color-highlight));">A little encouragement...</b><br>

Place names add an additional level of difficulty to working with geospatial data. People may notice and care if you use the wrong name for a place, or if you mispronounce a common local name. When in doubt about how to pronounce a place name, it is always a good idea to ask!

While you won't need to speak any place names out loud for this module, Philadelphians pronounce Schuylkill as "SKOO-kill."

</div>

Address
-------

The next two descriptions, the addresses, are how a lot of location data is provided. Addresses can convey a lot of information, but can also obscure important relations, like the fact that it is much easier to enter the building, at least by foot, from the South St. bridge. The street address doesn't tell you anything about the building's location in relation to anywhere not on Schuylkill Avenue, nor does it tell you where to find Schuylkill Avenue. The mailing address gives a little more information by specifying that the building is located inside the 19146 zip code, but that could be a fairly large geographical area.

Precise Coordinates
-------
The last description, which gives the latitude and longitude of the building, pinpoints the exact location on the planet. The coordinates are enough to place the building precisely on a map, or take a picture of it from a satellite. If you had the latitude and longitude of another building it would be easy to calculate the distance between the two. This is why having latitude and longitude coordinates can be so important when using location data in research.

> On the other hand, the precise coordinates leave out some important information that the building's address might provide. Precise distance, for example, turns out to be less relevant for calculating travel times when you know that the part of Schuylkill Avenue on which it sits is a one way street. The best location data would ideally contain both addresses and coordinates.

Author: Oh no, as soon as I saw this comment I remembered that I lived in Philadelphia for a few years before I was comfortable pronouncing that word!!! I'll add a pronunciation guide the first time it appears, as well as a "care" box that being able to say it is totally irrelevant to non-Philadelphians.

> On the other hand, the precise coordinates leave out some important information that the building's address might provide. Precise distance, for example, turns out to be less relevant for calculating travel times when you know that the part of Schuylkill Avenue on which it sits is a one way street. The best location data would ideally contain both addresses and coordinates.

Reviewer 2: i really tried different methods of pronunciation and non of them sounded right so I had to research it😅

The **equator** is the circle around the surface of the earth that is exactly halfway between the north and south poles. Any celestial body, like a planet or a star, that spins on an axis has two poles and an equator. Imagine a flat surface containing the equator and dividing the planet into a Northern Hemisphere and Southern Hemisphere. That flat surface is called the **equatorial plane**.

A **meridian** is a straight line on the surface of the earth connecting the poles. The third point we need is the location of the [Greenwich Observatory](https://www.rmg.co.uk/royal-observatory) in London, England. The meridian running through the observatory is called the **Prime Meridian**. Unlike the poles, which are fixed points determined by the astronomical fact that the earth is spinning on an axis, the location of the Prime Meridian was a human decision made for [historical](https://greenwichmeantime.com/articles/history/navy/) rather than geographical reasons.

Now that we have established the poles, the equator, and the prime meridian, we can use those to to create a coordinate system that will specify and point on earth!

> ![On the left an image of Earth is shown from the side with the North Pole labeled at the top, the South Pole labeled at the bottom, and the equator represented as a dashed horizontal line around the middle. On the right Earth is shown from above so that the North Pole is visible. The prime meridian is represented by a dashed vertical line starting at the North Pole and continuing down through the United Kingdom, the western Mediterranean Sea, and West Africa.](media/poles_meridian_equator.svg)<!-- style = "max-width:800px;" border = 5px solid -->

Author: You are absolutely right, and these images are some of the ones that I need a lot of help with. My usual go to person for these questions will be on vacation next week, so let's try to figure out some better wording together!

The goal of the alt text (as I understand it, please correct me if you have a different understanding) is to provide someone using a screen reader with the same content that the image provides. I have struggled to keep alt text reasonable length while making sure that the important information is available to people who can't see the picture.

Let's find a time next week to brainstorm some better alt text!

> ![On the left an image of Earth is shown from the side with the North Pole labeled at the top, the South Pole labeled at the bottom, and the equator represented as a dashed horizontal line around the middle. On the right Earth is shown from above so that the North Pole is visible. The prime meridian is represented by a dashed vertical line starting at the North Pole and continuing down through the United Kingdom, the western Mediterranean Sea, and West Africa.](media/poles_meridian_equator.svg)<!-- style = "max-width:800px;" border = 5px solid -->

Reviewer 2: i would love to brainstorm with you. The alt text from the description joy gave me during one of our one-on-one meetings, is supposed to be for those who cannot access/view the image for any reason

### Latitude and Longitude

> **Latitude** is a measure of how far north or south a location is from the equator. Given a point on the surface of the earth, imagine a straight line to the center of the earth. The angle that line makes with the equatorial plane, measured in degrees (denoted $ {}^\circ $), is the point's latitude. Since that angle could be either above or below the equatorial plane, we have to specify which angle we are talking about. The horizontal line on the picture below corresponding to 30 degrees north, or +30$ {}^\circ $, goes through the United States, while the line representing 30 degrees south, or -30$ {}^\circ $, runs through Chile and Argentina.

Author: Good call! What do you think about giving the degree symbol its own sentence?
```suggestion
**Latitude** is a measure of how far north or south a location is from the equator. Given a point on the surface of the earth, imagine a straight line to the center of the earth. The angle that line makes with the equatorial plane, measured in degrees, is the point's latitude. The symbol for degrees is the superscript circle $ {}^\circ $. 

Since the angle could be either above or below the equatorial plane, we have to specify which angle we are talking about. The horizontal line on the picture below corresponding to 30 degrees north, or +30$ {}^\circ $, goes through the United States, while the line representing 30 degrees south, or -30$ {}^\circ $, runs through Chile and Argentina.
```

> **Latitude** is a measure of how far north or south a location is from the equator. Given a point on the surface of the earth, imagine a straight line to the center of the earth. The angle that line makes with the equatorial plane, measured in degrees (denoted $ {}^\circ $), is the point's latitude. Since that angle could be either above or below the equatorial plane, we have to specify which angle we are talking about. The horizontal line on the picture below corresponding to 30 degrees north, or +30$ {}^\circ $, goes through the United States, while the line representing 30 degrees south, or -30$ {}^\circ $, runs through Chile and Argentina.

Reviewer 2: I think giving it its own sentence helps a lot

<div class = "important">
<b style="color: rgb(var(--color-highlight));">Important note</b><br>
Accurate geocoding can still result in slightly different coordinates for the same location. When analyzing geospatial data it is important to be aware of this possible "fuzziness" when considering whether two places are the same.
</div>

Multiple Addresses
----

Geocoding addresses one by one might be fine if you only have a few to do, but what if you have hundreds or even thousands of addresses to geocode? 

<div class = "warning">
<b style="color: rgb(var(--color-highlight));">Warning!</b><br>

When geocoding addresses for research, make sure that you are preserving the privacy and security of you participants. If you will be using addresses as part of research, you should have a plan for geocoding approved by the appropriate regulatory bodies before you begin.

You might need a formal agreement with a geocoding service (see the [Texas A&M Geoservice's privacy and security page](https://geoservices.tamu.edu/About/Legal/)) or a plan to do all geocoding locally on a computer owned by your institution. 

</div>

> If you have a public dataset, there are a number of geocoding services in existence, most of which will let you geocode addresses in batches. Google and Bing both have such services, and a more extensive list, along with links to the services, is available from [GISGeography.com](https://gisgeography.com/geocoders/).

Reviewer 2: ```suggestion
If you have a public dataset, there are a number of geocoding services in existence, most of which will let you geocode addresses in batches. Google and Bing both have such services. A more extensive list, along with links to the services, is available from [GISGeography.com](https://gisgeography.com/geocoders/).
```
The wording for the sentence was a little confusing.

While using these tools is outside the scope of this module, many are free to try out, especially if you will be geocoding a relatively small number of addresses (e.g. only a few thousand).

## Quiz

TRUE or FALSE: Addresses are irrelevant for research into the social determinants of health if you have access to latitude and longitude coordinates.

[( )] True
[(X)] False
***

<div class = "answer">

> Latitude and Longitude coordinates can tell you a lot that addresses can't, but they do not replace the important social and political information contained in a street address. Two people living geographically very close together but under different governmental regimes may have completely different healthcare options. Similarly, two people living seemingly far apart from each other but in the same town may both be impacted by the same municipal water source.

Author: No, I think that means I need to explain the word or avoid it! How does this sound?

```suggestion
Latitude and Longitude coordinates can tell you a lot that addresses can't, but they do not replace the important social and political information contained in a street address. Two people living geographically very close together but under different governmental regimes may have completely different healthcare options. Similarly, two people living seemingly far apart from each other but in the same town may both be impacted by the same city or district water source.
```

> Latitude and Longitude coordinates can tell you a lot that addresses can't, but they do not replace the important social and political information contained in a street address. Two people living geographically very close together but under different governmental regimes may have completely different healthcare options. Similarly, two people living seemingly far apart from each other but in the same town may both be impacted by the same municipal water source.

Reviewer 2: Funny story: After seeing the word here and researching it, I noticed the word everywhere. I was reading a paper and the word just popped up out of nowhere. It popped up about 5 times actually. 

I also agree with the change

> Latitude and Longitude coordinates can tell you a lot that addresses can't, but they do not replace the important social and political information contained in a street address. Two people living geographically very close together but under different governmental regimes may have completely different healthcare options. Similarly, two people living seemingly far apart from each other but in the same town may both be impacted by the same municipal water source.

Author: That happens to me a lot with new words and especially new idioms/cultural references!
